src/app.jsx

```diff
@@ -103,7 +103,17 @@ export const Application = () => {
             });
 
             channel.current.addEventListener("ready", () => {
-                setFiles(files);
+                Promise.all(files.map(file => {
+                    return cockpit.spawn(["ls", "-ld", "/" + path.join("/") + "/" + file.path], { superuser: "try" }).then(res => {
```

Comment:
You can use the stat command to get the octal permissions, since i see you convert these anyway.

`stat -c '%a' $yourfile`